evap/staff/templates/staff_semester_view.html

```diff
@@ -485,6 +485,27 @@ <h3 class="d-inline">{{ semester.name }}</h3>
                     </div>
                 </div>
                 {% if courses %}
+                    <form id="course-deletion-form" method="POST" action="{% url 'staff:course_delete' %}">
+                        {% csrf_token %}
+                    </form>
+                    <script defer>
+                        const form = document.getElementById("course-deletion-form");
+                        form.addEventListener("submit", event => {
+                            event.preventDefault();
+                            const body = new FormData(form);
+                            const request = fetch(
+                                form.action,
+                                { method: form.method, body },
+                            );
```

Comment:
I thought it's nice that the data flow is in the HTML and not in a script, but I agree that it is maybe more complicated that it needs to be.

I kind of forgot about this, but I wrote the form-agnostic code to see if it would maybe make sense to add a more general "async-submit-form` at some point (I think we have this pattern in a couple of places), which would reduce the code for this form to
```javascript
const form = ...;
form.addEventListener("post-submit", event => {
    fadeOutThenRemove(`#course-row-${event.detail.body.get("courseId")}`);
});
```

I will re-evaluate whether I like the `async-form` and see if the code gets any nicer

Reply:
I added some `custom-success` stuff